### PR TITLE
Terragrunt validate PR test & pre commit 

### PR DESCRIPTION
**Pull request by jjgriff93**
- [X] Adds Terragrunt (terraform) validation of all modules as part of pre-commit and PR tests
- [X] Fixes tflint so it actually does something (previously it wasn't actually running properly)
- [X] Implements terragrunt tflint pre-hook so we're aware of issues earlier on in lifecycle
- [X] Adds `hclfmt` to pre-commit which automatically fixes non-tf HCL files (terragrunt etc.)
- [X] Fixes all of the formatting errors we had going under the radar

### CODE REVIEW

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4745320317 (with refid `38975f74`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/264#issuecomment-1514969908) from @jjgriff93)

File: apps/app/locals.tf [hunk 1]
```diff
@@ -42,18 +42,18 @@ locals {
 
   # Map deployment branch and github environment names for main & testing slot (if enabled)
   branches_and_envs = var.app_config.add_testing_slot ? {
-    "${local.core_branch_name}"    = local.core_gh_env
-    "${local.testing_branch_name}" = local.testing_gh_env
-  } : { "${local.core_branch_name}" = local.core_gh_env }
+    local.core_branch_name    = local.core_gh_env
```
Comment (t-young31) [inline apps/app/locals.tf:45]: Does this work? I seem to remember it needing something like
```suggestion
    string(local.core_branch_name)    = local.core_gh_env
```
in pretty much any map

Comment (jjgriff93) [reply apps/app/locals.tf:45]: Works just fine, tested a deployment before raising this :) also validate would catch formatting things now hopefully!

Comment (jjgriff93) [reply apps/app/locals.tf:45]: Right you are as per the latest run :/ disappointed that validate doesn't catch this

Comment (jjgriff93) [reply apps/app/locals.tf:45]: Interestingly tflint invoked by terragrunt apply does when ran in the module folder - I shall investigate why pre-commit doesn't pick this up

```
vscode ➜ /workspaces/FlowEHR/apps (jjgriff93/terragrunt-validate-pre-commit) $ terragrunt apply
INFO[0001] Executing hook: before_hook                  
Plugin `terraform` is already installed
Plugin `azurerm` is already installed
Failed to check ruleset; Failed to check `azurerm_container_registry_webhook_invalid_name` rule: app/locals.tf:47,9-31: Ambiguous attribute key; If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap it in quotes to create a string literal., and 14 other diagnostic(s)
ERRO[0001] Error running hook before_hook with message: Error while running tflint with args: [tflint --init --config /workspaces/FlowEHR/.tflint.hcl /workspaces/FlowEHR/apps] 
ERRO[0001] Errors encountered running before_hooks. Not running 'terraform'. 
ERRO[0001] 1 error occurred:
        * Error while running tflint with args: [tflint --init --config /workspaces/FlowEHR/.tflint.hcl /workspaces/FlowEHR/apps]
 
ERRO[0001] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

**Review by jedheaj314 — APPROVED** [review]

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4755382900 (with refid `38975f74`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/264#issuecomment-1516415033) from @jjgriff93)

**Pull request merged**